compiler/rustc_codegen_cranelift/src/abi/mod.rs

```diff
@@ -146,15 +146,17 @@ impl<'tcx> FunctionCx<'_, '_, 'tcx> {
             (params, args.into())
         };
 
-        // Return i128 using a return area pointer on Windows and s390x.
-        let adjust_ret_param =
-            if self.tcx.sess.target.is_like_windows || self.tcx.sess.target.arch == "s390x" {
-                returns.len() == 1 && returns[0].value_type == types::I128
-            } else {
-                false
-            };
-
-        if adjust_ret_param {
+        // Return i128 using the vector ABI on Windows
+        let ret_single_i128 = returns.len() == 1 && returns[0].value_type == types::I128;
+        if ret_single_i128 && self.tcx.sess.target.is_like_windows {
+            returns[0].value_type = types::I64X2;
+            let ret = self.lib_call_unadjusted(name, params, returns, &args)[0];
+            // FIXME(bytecodealliance/wasmtime#6104) use bitcast instead of store to get from i64x2 to i128
+            let ret_ptr = self.create_stack_slot(16, 16);
+            ret_ptr.store(self, ret, MemFlags::trusted());
+            Cow::Owned(vec![ret_ptr.load(self, types::I128, MemFlags::trusted())])
```

Comment:
```suggestion
            returns[0].value_type = types::I64X2;

            let ret = self.lib_call_unadjusted(name, params, returns, &args)[0];
            
            // FIXME(bytecodealliance/wasmtime#6104) use bitcast instead of store to get from i64x2 to i128
            let ret_ptr = self.create_stack_slot(16, 16);
            ret_ptr.store(self, ret, MemFlags::trusted());
            Cow::Owned(vec![ret_ptr.load(self, types::I128, MemFlags::trusted())])
```